java/com/google/security/wycheproof/testcases/AesGcmTest.java

```diff
@@ -268,6 +270,167 @@ public void testByteBufferAlias() throws Exception {
     }
   }
 
+  /** Encryption and decryption with large arrays should be copy-safe. */
+  public void testLargeArrayAlias() throws Exception {
+    byte[] ptVector = new byte[8192];
+
+    // this offset is relative to the start of the input, not the start of the buffer.
+    for (int outputOffset = -32; outputOffset <= 32; outputOffset++) {
+      // try with doFinal directly as well as with update followed by doFinal
+      for (int useUpdate = 0; useUpdate <= 1; useUpdate++) {
+        SecretKeySpec key = new SecretKeySpec(new byte[16], "AES");
+        GCMParameterSpec parameters = new GCMParameterSpec(128, new byte[16]);
+        Cipher cipher = Cipher.getInstance("AES/GCM/NoPadding");
+        cipher.init(Cipher.ENCRYPT_MODE, key, parameters);
+
+        // these offsets are relative to the start of the buffer
+        int inputOffsetInBuffer = 32;
+        int outputOffsetInBuffer = inputOffsetInBuffer + outputOffset;
+        int sliceLength = cipher.getOutputSize(ptVector.length);
+
+        byte[] inBuf = new byte[sliceLength + Math.max(inputOffsetInBuffer, outputOffsetInBuffer)];
+        byte[] outBuf = inBuf;
+
+        System.arraycopy(ptVector, 0, inBuf, inputOffsetInBuffer, ptVector.length);
+
+        try {
+          int ctLength = 0;
+          if (useUpdate > 0) {
+            ctLength += cipher.update(inBuf, inputOffsetInBuffer, ptVector.length, outBuf, outputOffsetInBuffer);
+            ctLength += cipher.doFinal(inBuf, 0, 0, outBuf, outputOffsetInBuffer + ctLength);
+          } else {
+            ctLength += cipher.doFinal(inBuf, inputOffsetInBuffer, ptVector.length, outBuf, outputOffsetInBuffer);
+          }
+
+          System.arraycopy(outBuf, outputOffsetInBuffer, inBuf, inputOffsetInBuffer, ctLength);
+
+          cipher = Cipher.getInstance("AES/GCM/NoPadding");
+          cipher.init(Cipher.DECRYPT_MODE, key, parameters);
+
+          int resultPtLength = 0;
+          if (useUpdate > 0) {
+            resultPtLength += cipher.update(inBuf, inputOffsetInBuffer, ctLength, outBuf, outputOffsetInBuffer);
+            resultPtLength += cipher.doFinal(inBuf, 0, 0, outBuf, outputOffsetInBuffer + resultPtLength);
+          } else {
+            resultPtLength += cipher.doFinal(inBuf, inputOffsetInBuffer, ctLength, outBuf, outputOffsetInBuffer);
+          }
+
+          assertEquals(resultPtLength, ptVector.length);
+          assertArrayEquals(ptVector,
+                            Arrays.copyOfRange(outBuf, outputOffsetInBuffer, outputOffsetInBuffer + resultPtLength));
+        } catch (Throwable t) {
+          throw new AssertionError("testLargeByteBufferAlias failed with outputOffset=" + outputOffset, t);
+        }
+      }
+    }
+  }
+
+  /**
```

Comment:
Thanks for your pull request. Can you please add https://bugs.openjdk.java.net/browse/JDK-8181386 which has more detailed description of the bug to the comment?